### DISCUSSION TITLE
Why two `size_of_entry` tests for `gix_index`?

### DISCUSSION THREAD

**Discussion by EliahKagan** [GitoxideLabs/gitoxide#1685] [Q&A] [answered]
This is the `gix-index size_of_entry` test:

https://github.com/GitoxideLabs/gitoxide/blob/9ab86a23d45941c4f0a3239e0cb57d4161dd279c/gix-index/src/lib.rs#L241-L248

While is the `gix-index-tests::integrate index::size_of_entry` test:

https://github.com/GitoxideLabs/gitoxide/blob/9ab86a23d45941c4f0a3239e0cb57d4161dd279c/gix-index/tests/index/mod.rs#L26-L33

Is there a need for this pair of separate but identical tests? (They are effectively identical because `crate` in the first test refers to `gix_index`, which appears in its place in the second test.)

---

*Edit:* There is a similar situation in `gix-pack`, where the pair of tests that test `Item<()>` and `Item<EntryWithDefault>` are repeated in two places.

These are the `gix-pack cache::delta::tests::size_of_pack_tree_item` and `gix-pack cache::delta::tests::size_of_pack_verify_data_structure` tests:

https://github.com/GitoxideLabs/gitoxide/blob/9ab86a23d45941c4f0a3239e0cb57d4161dd279c/gix-pack/src/cache/delta/mod.rs#L29-L33

https://github.com/GitoxideLabs/gitoxide/blob/9ab86a23d45941c4f0a3239e0cb57d4161dd279c/gix-pack/src/cache/delta/mod.rs#L35-L49

While these are the `gix-pack cache::delta::tree::tests::size_of_pack_tree_item` and `gix-pack cache::delta::tree::tests::size_of_pack_verify_data_structure` tests:

https://github.com/GitoxideLabs/gitoxide/blob/9ab86a23d45941c4f0a3239e0cb57d4161dd279c/gix-pack/src/cache/delta/tree.rs#L229-L233

https://github.com/GitoxideLabs/gitoxide/blob/9ab86a23d45941c4f0a3239e0cb57d4161dd279c/gix-pack/src/cache/delta/tree.rs#L235-L249

**Comment by Byron** [comment, marked as answer]
Thanks for bringing this up!

And even though I could have thought that maybe I had the same idea, twice, in the case of `gix-index`, it's hard to imagine that happened with the much more complex duplicated tests in `gix-pack`.
 
My answer is that this must have been some sort of accident causing the duplication, I wouldn't see a reason them to exist twice.

**Reply by EliahKagan** [reply]
For the `gix-pack` tests, looking again, it looks like c85a981 (#1237) created the `tree` module, moved code into it, and this one associated test ended up in both places. I don't know what the cause was in `gix-index`, though.

For both `gix-index` and `gix-pack`, I'll deduplicate them as part of a soon-to-be PR.

**Reply by EliahKagan** [reply]
I've included the removal of one of each of the duplicated tests in #1687.
